packages/core/src/rum/__tests__/DdRum.test.ts

```diff
@@ -452,33 +454,273 @@ describe('DdRum', () => {
+                it('exposes TracingIdType enum', () => {
+                    expect(TracingIdType).toBeDefined();
+                });
             });
 
-            it('generates a valid span id in decimal format', () => {
-                const uuid = DdRum.generateUUID(TracingIdType.span);
+            describe('Default behavior with invalid input', () => {
+                it('generates 64-bit span id when passed invalid tracing id type', () => {
+                    const randomPropagatorType = Object.values(PropagatorType);
+
+                    for (let i = 0; i < 100; i++) {
+                        const tempContextPropagator =
+                            randomPropagatorType[
+                                Math.floor(
+                                    Math.random() * randomPropagatorType.length
+                                )
+                            ];
+
+                        const uuid = DdRum.generateUUID(
+                            ('wrong' as unknown) as TracingIdType
+                        );
+
+                        const uuidPropagator = uuid.forContextPropagation(
+                            tempContextPropagator
+                        );
+
+                        expect(uuidPropagator).toBeDefined();
+
+                        if (tempContextPropagator === PropagatorType.DATADOG) {
+                            expect(
+                                TracingIdentifierUtils.isWithin64Bits(
+                                    uuidPropagator
+                                )
+                            ).toBe(true);
+                        } else {
+                            expect(uuidPropagator).toMatch(/^[0-9a-f]{16}$/);
+                            expect(
+                                TracingIdentifierUtils.isWithin64Bits(
+                                    uuidPropagator,
+                                    16
+                                )
+                            ).toBe(true);
+                        }
+
+                        const uuidDecimal = uuid.forResource();
+                        expect(uuidDecimal).toBeDefined();
+                        expect(
+                            TracingIdentifierUtils.isWithin64Bits(uuidDecimal)
+                        ).toBe(true);
+                    }
+                });
+            });
+
+            describe('Trace ID generation', () => {
+                it('generates valid 128-bit trace ID in decimal format for Datadog propagator', () => {
+                    let iterations = 100;
+                    while (iterations-- > 0) {
+                        const uuidObject = DdRum.generateUUID(
+                            TracingIdType.trace
+                        );
+
+                        const uuidStrLow64Propagator = uuidObject.forContextPropagation(
+                            PropagatorType.DATADOG
+                        );
```

Comment:
shouldn't `forContextPropagation` return 2 chunks: low 64 bits and high 64 bits?